dev/AnimatedVisualPlayer/AnimatedVisualPlayer.cpp

```diff
@@ -755,6 +780,75 @@ void AnimatedVisualPlayer::OnAutoPlayPropertyChanged(
     }
 }
 
+void AnimatedVisualPlayer::OnAnimationOptimizationPropertyChanged(
+    winrt::DependencyPropertyChangedEventArgs const& args)
+{
+    auto optimization = unbox_value<winrt::PlayerAnimationOptimization>(args.NewValue());
+
+    if (m_nowPlaying)
+    {
+        // If there is something in play right now we should not create/destroy animations.
+        return;
+    }
+
+    if (optimization == winrt::PlayerAnimationOptimization::Resources)
+    {
+        DestroyAnimations();
+    }
+    else if (optimization == winrt::PlayerAnimationOptimization::Latency)
+    {
+        CreateAnimations();
+    }
+}
+
+void AnimatedVisualPlayer::CreateAnimations() {
+    m_createAnimationsCounter++;
+
+    if (m_isAnimationsCreated) 
+    {
+        return;
+    }
+
+    // Check if current animated visual supports creating animations and create them.
+    if (const auto& animatedVisual = m_animatedVisual.get())
+    {
+        if (const auto& animatedVisual2 = m_animatedVisual.try_as<winrt::IAnimatedVisual2>())
+        {
+            animatedVisual2.CreateAnimations();
+            m_isAnimationsCreated = true;
+        }
+    }
+}
+
+void AnimatedVisualPlayer::DestroyAnimations() {
+    if (!m_isAnimationsCreated)
+    {
+        return;
+    }
+
+    // Call RequestCommit to make sure that previous compositor calls complete before destroying animations.
+    m_rootVisual.Compositor().RequestCommitAsync().Completed(
```

Comment:
RequestCommitAsync() is only available on Rs4+ we need to gracefully handle the case where this API isn't available on RS2 and RS3.